project3/coordinator/CMakeLists.txt

```diff
@@ -0,0 +1,2 @@
+add_library(coordinator connected_participants.cpp message_log.cpp registrar.cpp messenger.cpp)
+target_link_libraries(coordinator pub_sub_proto wire_protocol queue)
```

Comment:
I'm surprised this works, since there's no specific `queue` target. Since the queue is a header-only library, you don't have to specify any link libraries to use is.